network/src/filters/filter.rs

```diff
@@ -32,6 +32,13 @@ impl Default for Filter {
 }
 
 impl Filter {
+    pub fn new(input_vector: Vec<IpAddr>) -> Self {
+        Self {
+            enabled: false,
```

Comment:
or how about `enabled: !input_vector.is_empty()`

Reply:
It sounds good. It might be better than adding more arguments for enabling/disabling.